### ISSUE TITLE
Remove the legacy (v1 & v1.5) API

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#9274]
Firefox60+ uses the v3 API; previous Firefoxes used `/api/1.5/` (and other non-Firefox toolkit apps, and a possible unknown number of 3rd party users).  After the support period ends for Firefox52 ESR we can remove the v1 and v1.5 API from addons-server.

The MDN [page](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/AMO/Add-ons_manager_API) has said it's deprecated for years.

**Comment by EnTeQuAk** [comment]
Looking at datadog, we'll probably want to be careful and ask ops ahead of removal to check what user agents are actually requesting this resource.

![screenshot from 2018-09-18 13-54-26](https://user-images.githubusercontent.com/139033/45685685-70d85580-bb4a-11e8-8c3d-92c8b57970ca.png)

We have around ~4k requests per day on our legacy api.

**Comment by EnTeQuAk** [comment]
@bqbn @autrilla are you able to get some stats about the kind of request, e.g user agents that are hitting `/api/v1/` and `/api/v1.5/`? They all should happen on `services.addons.mozilla.org` afaik

**Comment by eviljeff** [comment]
https://gist.github.com/autrilla/3abc812277bc4cab75dc512aadc6dcda for the top 1000 user agents hitting the `/api/1.5/search` endpoint yesterday (2018-11-01).  There's nothing too surprising in there - it's old versions of Firefox & Thunderbird (pre-60).

**Comment by eviljeff** [comment]
@jvillalobos is there anything in that list that stops us from removing the 1.5 api?  If so, what would be the future trigger/milestone to allow us to remove it.

**Comment by jvillalobos** [comment]
52.9.1 is the current release version of Thunderbird. Shouldn't those clients be hitting the API at addons.thunderbird.net?

I also see lots of Firefox 52 requests, which is the previous ESR. I would prefer to defer this and measure it again early in Q1, to see if disabling legacy versions and people being updated to the current ESR change the usage of the old API significantly.

**Comment by eviljeff** [comment]
> 52.9.1 is the current release version of Thunderbird. Shouldn't those clients be hitting the API at addons.thunderbird.net?

Release is on [60.3.0 now](https://www.thunderbird.net/en-US/thunderbird/releases/).  In general though, we can't control what url Thunderbird does or doesn't hit; and after we removed all the Thunderbird compatible versions they'll be getting empty responses anyway.

**Comment by jvillalobos** [comment]
Hmm, my profile was stuck in 52 and didn't update for some reason, so maybe that's why there are so many 52 users. Like you said, though, they aren't getting anything useful back, so it doesn't make a difference.

**Comment by EnTeQuAk** [comment]
![screenshot from 2019-01-16 09-10-02](https://user-images.githubusercontent.com/139033/51234974-977a9180-196e-11e9-8c9f-c326b4651f7c.png)

**Comment by EnTeQuAk** [comment]
@autrilla any chance you could fetch another report like https://gist.github.com/autrilla/3abc812277bc4cab75dc512aadc6dcda for the top 1000 user agents hitting the /api/1.5/search endpoint? That'd help understanding our current situation again.

**Comment by EnTeQuAk** [comment]
```
1140354 Mozilla/5.0 (Windows NT 5.1; rv:52.0) Gecko/20100101 Firefox/52.0
 271125 Mozilla/5.0 (Windows NT 6.0; rv:52.0) Gecko/20100101 Firefox/52.0
 157522 Mozilla/5.0 (X11; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0
 134938 Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0
 130053 Mozilla/5.0 (Macintosh; Intel Mac OS X 10.6; rv:48.0) Gecko/20100101 Firefox/48.0
 116396 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0
 100230 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:56.0) Gecko/20100101 Firefox/56.0
  96495 Mozilla/5.0 (Windows NT 6.1; rv:56.0) Gecko/20100101 Firefox/56.0
  83525 Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:57.0) Gecko/20100101 Firefox/57.0
  70381 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:56.0) Gecko/20100101 Firefox/56.0
  67723 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:47.0) Gecko/20100101 Firefox/47.0
  67707 Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0
  67140 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0
  64909 Mozilla/5.0 (Macintosh; Intel Mac OS X 10.7; rv:48.0) Gecko/20100101 Firefox/48.0
  63418 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:57.0) Gecko/20100101 Firefox/57.0
  60597 Mozilla/5.0 (Windows NT 6.1; rv:52.0) Gecko/20100101 Firefox/52.0
  58913 Mozilla/5.0 (Windows NT 6.1; rv:47.0) Gecko/20100101 Firefox/47.0
  57484 Mozilla/5.0 (Windows NT 6.0; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0
  56747 Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0
  51604 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0
  51301 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:43.0) Gecko/20100101 Firefox/43.0
  50570 Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0
  46957 Mozilla/5.0 (Windows NT 6.1; rv:43.0) Gecko/20100101 Firefox/43.0
  43173 Mozilla/5.0 (Windows NT 6.1; rv:57.0) Gecko/20100101 Firefox/57.0
  41675 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:47.0) Gecko/20100101 Firefox/47.0
  41590 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0
  41413 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:54.0) Gecko/20100101 Firefox/54.0
  39509 Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:48.0) Gecko/20100101 Firefox/48.0
  38098 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:57.0) Gecko/20100101 Firefox/57.0
  36502 Mozilla/5.0 (Windows NT 6.1; rv:59.0) Gecko/20100101 Firefox/59.0
  36406 Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36
  36210 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:50.0) Gecko/20100101 Firefox/50.0
  33236 Mozilla/5.0 (Windows NT 6.1; rv:54.0) Gecko/20100101 Firefox/54.0
  30746 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:45.0) Gecko/20100101 Firefox/45.0
  29762 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:53.0) Gecko/20100101 Firefox/53.0
  28936 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:51.0) Gecko/20100101 Firefox/51.0
  28222 Mozilla/5.0 (Android 8.0.0; Mobile; rv:56.0) Gecko/56.0 Firefox/56.0
  27660 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:43.0) Gecko/20100101 Firefox/43.0
  26839 Mozilla/5.0 (Windows NT 6.1; rv:53.0) Gecko/20100101 Firefox/53.0
  26529 Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:58.0) Gecko/20100101 Firefox/58.0
  26446 Mozilla/5.0 (Windows NT 6.2; rv:42.0) Gecko/20100101 Firefox/42.0
  25997 Mozilla/5.0 (Windows NT 5.1; rv:43.0) Gecko/20100101 Firefox/43.0
  25481 Mozilla/5.0 (Windows NT 5.1; rv:48.0) Gecko/20100101 Firefox/48.0
  25037 Mozilla/5.0 (Windows NT 5.1; rv:47.0) Gecko/20100101 Firefox/47.0
  24867 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:54.0) Gecko/20100101 Firefox/54.0
  24709 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:48.0) Gecko/20100101 Firefox/48.0
  24084 Mozilla/5.0 (Windows NT 6.1; rv:45.0) Gecko/20100101 Firefox/45.0
  23930 Mozilla/5.0 (X11; Linux i686; rv:38.0) Gecko/20100101 Firefox/38.0
  23686 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0
  23289 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:49.0) Gecko/20100101 Firefox/49.0
  23108 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:56.0; Waterfox) Gecko/20100101 Firefox/56.2.6
  23063 Mozilla/5.0 (Windows NT 6.3; rv:36.0) Gecko/20100101 Firefox/36.0
  22925 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:52.0) Gecko/20100101 Firefox/52.0
  22538 Mozilla/5.0 (Windows NT 6.1; rv:48.0) Gecko/20100101 Firefox/48.0
  21474 Mozilla/5.0 (X11; Linux x86_64; rv:49.0) Gecko/20100101 Firefox/49.0
  21303 Mozilla/5.0 (Windows NT 6.3; WOW64; rv:56.0) Gecko/20100101 Firefox/56.0
  21121 Mozilla/5.0 (Windows NT 6.1; rv:49.0) Gecko/20100101 Firefox/49.0
  21109 Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:54.0) Gecko/20100101 Firefox/54.0
  20707 Mozilla/5.0 (Windows NT 6.1; rv:51.0) Gecko/20100101 Firefox/51.0
  20386 Mozilla/5.0 (Windows NT 6.1; rv:58.0) Gecko/20100101 Firefox/58.0
  20379 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:37.0) Gecko/20100101 Firefox/37.0
  20344 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:48.0) Gecko/20100101 Firefox/48.0
  20240 Mozilla/5.0 (Windows NT 6.1; rv:50.0) Gecko/20100101 Firefox/50.0
  19626 Mozilla/5.0 (X11; Ubuntu; Linux i686; rv:52.0) Gecko/20100101 Firefox/52.0
  19301 Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0
  19188 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:45.0) Gecko/20100101 Firefox/45.0
  19139 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:58.0) Gecko/20100101 Firefox/58.0
  18679 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:41.0) Gecko/20100101 Firefox/41.0
  18380 Mozilla/5.0 (Windows NT 6.3; Win64; x64; rv:57.0) Gecko/20100101 Firefox/57.0
  18041 Mozilla/5.0 (Windows NT 6.3; WOW64; rv:48.0) Gecko/20100101 Firefox/48.0
  17930 Mozilla/5.0 (Windows NT 5.1; rv:52.0) Gecko/20100101 Thunderbird/52.9.1 Lightning/5.4.9.1
  17918 Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:52.0) Gecko/20100101 Firefox/52.0
  17803 Mozilla/5.0 (Windows NT 5.2; rv:52.0) Gecko/20100101 Firefox/52.0
  17595 Mozilla/5.0 (Windows NT 6.3; WOW64; rv:47.0) Gecko/20100101 Firefox/47.0
  17251 Mozilla/5.0 (Windows NT 6.3; WOW64; rv:41.0) Gecko/20100101 Firefox/41.0
  17216 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:51.0) Gecko/20100101 Firefox/51.0
  17076 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:59.0) Gecko/20100101 Firefox/59.0
  16636 Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:59.0) Gecko/20100101 Firefox/59.0
  15653 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:50.0) Gecko/20100101 Firefox/50.0
  15596 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:52.0) Gecko/20100101 Thunderbird/52.9.1 Lightning/5.4.9.1
  15488 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:46.0) Gecko/20100101 Firefox/46.0
  15423 Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0
  15301 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:53.0) Gecko/20100101 Firefox/53.0
  15292 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:57.0) Gecko/20100101 Firefox/57.0
  15009 Mozilla/5.0 (Windows NT 6.3; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0
  14292 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:42.0) Gecko/20100101 Firefox/42.0
  13567 Mozilla/5.0 (Windows NT 6.1; rv:44.0) Gecko/20100101 Firefox/44.0
  13483 Mozilla/5.0 (Windows NT 5.1; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0
  13142 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:55.0) Gecko/20100101 Firefox/55.0
  12936 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:49.0) Gecko/20100101 Firefox/49.0
  12916 Mozilla/5.0 (Windows NT 10.0; WOW64; rv:45.0) Gecko/20100101 Thunderbird/45.8.0 Lightning/4.7.8
  12803 Mozilla/5.0 (Windows NT 6.1; rv:38.0) Gecko/20100101 Firefox/38.0
  12735 Mozilla/5.0 (Linux; <Android Version>; <Build Tag etc.>) AppleWebKit/<WebKit Rev> (KHTML, like Gecko) Chrome/<Chrome Rev> Mobile Safari/<WebKit Rev>
  12672 Mozilla/5.0 (X11; Linux i686; rv:45.0) Gecko/20100101 Firefox/45.0
  12660 Mozilla/5.0 (X11; Linux i686; rv:52.0) Gecko/20100101 Firefox/52.0
  11908 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:44.0) Gecko/20100101 Firefox/44.0
  11796 Mozilla/5.0 (iPhone; U; CPU iPhone OS 4_0 like Mac OS X; en-us) AppleWebKit/532.9 (KHTML, like Gecko) Version/4.0.5 Mobile/8A293 Safari/6531.22.7
  11775 Mozilla/5.0 (Android 7.0; Mobile; rv:56.0) Gecko/56.0 Firefox/56.0
11659 Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.0
```

**Comment by EnTeQuAk** [comment]
More data: https://irccloud.mozilla.com/pastebin/uNrLq4tm/ (that's apparently since Jan 25th)

More: https://irccloud.mozilla.com/pastebin/eWS89ray/

![screenshot from 2019-02-08 12-47-35](https://user-images.githubusercontent.com/139033/52476792-10e95680-2ba0-11e9-8d34-97ed409209a3.png)

So… it's pretty much unused and really should safe to remove.

Bonus points to double check that `about:addons` etc still works for Firefox 52.

**Comment by EnTeQuAk** [comment]
QA: This is removing the legacy api that has been used until (and including) Firefox 52, e.g for retrieving detail information about installed add-ons and other add-ons related functionality.

I tested Firefox 52 ESR locally all installed add-ons continued to work, only detailed information wasn't available (or only partially) but it didn't break.

This does not impact updating installed add-ons, so Firefox 52 ESR users continue to receive add-on updates if necessary.

**Comment by ioanarusiczki** [comment]
@EnTeQuAk I verified this with FF52.9.0 ESR on AMO dev (Win10). 
- I could install then update versions on FF52.9.0 ESR with Check for Updates and restarting the browser.
- I installed versions on FF65 then opened the same profile with ESR and updated versions successfully (in this case the ESR will require a restart every time).

![a restart is required](https://user-images.githubusercontent.com/33448286/53093671-c0b7b000-3520-11e9-8a93-9ac379f8ca15.gif)

- I installed versions on ESR then updated them with FF65 successfully.
- for the add-ons I've tested with the information was available.
